### PR TITLE
Add 'catalog' to Quilt+ URIs

### PR DESCRIPTION
**Pull request by drernie**
# Description

This is my formal request to allow specifying the Catalog as part of the Quilt+ URI. This will allow applications that take a Quilt+ URI (nf-quilt, QuiltSync) to generate and share human-readable URLs.

Adding it as a fragment appears backwards-compatible; adding it as a query string, on the other hand, breaks the resolver.

### CODE REVIEW

**Review by nl0 — COMMENTED** [review]
i feel somewhat ambivalent about that proposal, but i think i can live with this

File: catalog/app/utils/PackageUri.ts [hunk 1]
```diff
@@ -108,5 +108,6 @@ export function stringify({ bucket, name, hash, tag, path }: PackageUri) {
     pkgSpec += `:${tag}`
   }
   const pathPart = path ? `&path=${encodeURIComponent(path)}` : ''
-  return `quilt+s3://${bucket}#package=${pkgSpec}${pathPart}`
+  const catalogPart = window.location.hostname
```
Comment (nl0) [inline catalog/app/utils/PackageUri.ts:111]: even tho it may seem like a nice shortcut, i deeply disagree with this approach.
i believe the `catalog` part should be added to the model and handled explicitly

File: docs/CHANGELOG.md [hunk 2]
```diff
@@ -13,6 +13,12 @@ Entries inside each section should be ordered by type:
 ## CLI
 !-->
 
+# unreleased - YYYY-MM-DD
+
+## Python API
+
+* [Changed] Add `catalog` fragment to Quilt+ URIs (and to documentation). ([#4213](https://github.com/quiltdata/quilt/pull/4213))
```
Comment (nl0) [inline docs/CHANGELOG.md:20]: how is this related to the python api?

**Review by nl0 — CHANGES_REQUESTED** [review]

File: docs/CHANGELOG.md (hunk 2 above)
Comment (drernie) [reply docs/CHANGELOG.md:20]: Doh! Dumb copy-and-paste error.

File: catalog/app/utils/PackageUri.ts (hunk 1 above)
Comment (drernie) [reply catalog/app/utils/PackageUri.ts]: Okay, I've interpreted `model` as "stringify takes it as an argument", which is passed from:
- Dashboard
- PackageTreeWrapper (on through to PackageCodeSample)

Is that what you meant? If not, any chance you could fix it for me so it makes this release?

**Comment by drernie** [comment]
@fiskus Can you help me out with the "informational" test coverage failure? I don't see any good examples of how to test React.

**Comment by fiskus** [comment]
We tend to ignore these failures.

I can write the test for `app/containers/Bucket/CodeSamples/Package.tsx`, but I'm afraid that writing tests for `app/containers/Bucket/PackageTree/PackageTree.tsx` would be too difficult.

File: catalog/app/utils/PackageUri.ts (hunk 1 above)
Comment (nl0) [reply catalog/app/utils/PackageUri.ts]: > Is that what you meant?

yep

> If not, any chance you could fix it for me so it makes this release?

ok ill try

File: docs/CHANGELOG.md (hunk 2 above)
Comment (nl0) [reply docs/CHANGELOG.md:20]: we have a separate changelog for the catalog

**Review by nl0 — APPROVED** [review]
